src/fread.c

```diff
@@ -767,11 +767,23 @@ static void parse_double_regular(FieldParseContext *ctx)
     }
     e += Eneg? -E : E;
   }
-  e += 350; // lookup table is arranged from -350 (0) to +350 (700)
-  if (e<0 || e>700) goto fail;
+  if (e<-350 || e>350) goto fail;
 
-  double r = (double)((long double)acc * pow10lookup[e]);
-  *target = neg? -r : r;
+  long double r = (long double)acc;
+  if (e < -300 || e > 300) {
+    // Handle extra precision by pre-multiplying the result by pow(10, extra),
+    // and then remove extra from e.
+    // This avoids having to store very small or very large constants that may
+    // fail to be encoded by the compiler, even though the values can actually
+    // be stored correctly.
```

Comment:
Very clear comment and wording. Perfect.
Many thanks for investigating and fixing this!